pkg/vsphere/tasks/waiter.go

```diff
@@ -91,15 +91,19 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 
 func isTaskInProgress(err error) bool {
 	if soap.IsSoapFault(err) {
-		if _, ok := soap.ToSoapFault(err).VimFault().(types.TaskInProgress); ok {
+		f := soap.ToSoapFault(err)
+		if _, ok := f.VimFault().(types.TaskInProgress); ok {
 			return true
 		}
+		log.Debugf("unexpected soap fault on task retry : %s", spew.Sdump(f))
```

Comment:
btw you should log `f.VimFault()` here, rather than `f` and `%#v` should be just fine.  If go-spew is helpful elsewhere, fine, just not seeing how it is with the provided example output.